protocol/kafka_sarama/v2/receiver.go

```diff
@@ -53,19 +53,25 @@ func (r *Receiver) Close(context.Context) error {
 }
 
 func (r *Receiver) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
-	for message := range claim.Messages() {
-		msg := message
-		m := NewMessageFromConsumerMessage(msg)
-
-		r.incoming <- msgErr{
-			msg: binding.WithFinish(m, func(err error) {
-				if protocol.IsACK(err) {
-					session.MarkMessage(msg, "")
-				}
-			}),
+	for {
+		select {
+		case msg, ok := <-claim.Messages():
+			if !ok {
+				return nil
```

Comment:
question: since the behavior is now different from the previous implementation (the previous one basically never returned IIUC), should we add a comment here to describe when `ConsumeClaim` returns and what `error` would be (if any)?

Reply:
Added a comment in code for this, please review.